packages/laverna/src/cli.js

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env node
+/* eslint-disable no-restricted-globals */
+/* eslint-disable n/shebang */
+// @ts-check
+
+const { bold, magenta, gray, italic } = require('kleur')
+const util = require('node:util')
+const { Laverna } = require('./laverna')
+const { ERR } = require('./log-symbols')
+const { name, bugs, version, description } = require('../package.json')
+
+/**
+ * CLI for `laverna`, which wraps `npm publish` to only attempt publishing
+ * packages that have not yet been published.
+ *
+ * @packageDocumentation
+ */
+
+/**
+ * Options for {@link util.parseArgs} config (`options` property).
+ *
+ * @satisfies {import('node:util').ParseArgsConfig['options']}
+ */
+const options = /**
+ * @type {const}
+ */ ({
+  dryRun: {
+    type: 'boolean',
+  },
+  'dry-run': {
+    type: 'boolean',
+  },
+  root: {
+    type: 'string',
+  },
+  help: {
+    type: 'boolean',
+  },
+  newPkg: {
+    type: 'string',
+    multiple: true,
+  },
+  'new-pkg': {
+    type: 'string',
+    multiple: true,
+  },
+  yes: {
+    type: 'boolean',
+    short: 'y',
+  },
+})
+
+/**
+ * Parses CLI args
+ *
+ * @param {string[]} [args]
+ */
+function parseArgs(args) {
+  const { values } = util.parseArgs({
+    args,
+    options,
+  })
+  return values
+}
+
+function main() {
+  let opts = parseArgs()
+
+  if (opts.help) {
+    // this should be the only place we write to stdout unless we
+    // want to start outputting JSON
+    console.log(`
+  ${bold('laverna')} [options..]
+
+  ${italic(description)}
+
+  Options:
+
+  --dryRun        - Enable dry-run mode
+  --root=<path>   - Path to workspace root (default: current working dir)
+  --newPkg=<name> - Workspace <name> has never been published (repeatable)
+  --yes/-y        - Skip confirmation prompt
+
+  Problems? Visit ${bugs.url}
+
+  `)
+  } else {
+    // refuse to run with both camelCase and kebab-case flags
+    if ((opts.dryRun && opts['dry-run']) || (opts.newPkg && opts['new-pkg'])) {
+      console.error(ERR, 'Use camelCase or kebab-case flags; not both')
+      process.exitCode = 1
+      return
+    }
+
+    // prefer camelCase
+    opts.dryRun = opts.dryRun ?? opts['dry-run']
+    opts.newPkg = opts.newPkg ?? opts['new-pkg']
```

Comment:
```suggestion
    // prefer camelCase
    opts.dryRun = opts.dryRun || opts['dry-run']
    opts.newPkg = new Set([..opts.newPkg, ...opts['new-pkg'])
```

- `--newPkg` / `--new-pkg` is repeatable. While a bit silly, I don't see why `--newPkg=foo --new-pkg=bar` should not be considered valid?

- being a boolean flag defaulting to `false` and not accepting passed values/negation, `--dryRun`, `--dry-run`, and `--dryRun --dry-run` should all be redundant/equivalent



So in both cases we don't really have notion of conflicts and need for precedence, so could prob skip these checks alltogether?

Reply:
I don't want to write code to merge and dedupe the arrays generated by passing `--newPkg` and `--new-pkg`.  I also didn't want inconsistent behavior around `--dryRun`.  

I didn't really want to write _any_ code to handle both kebab-case and camelCase.  If we need stuff like that, I'd rather just pull in `yargs`.

I'm inclined to rip the kebab-case stuff out.

Reply:
Another way to look at it: we can add all the bells & whistles later, if desired.  While I absolutely understand the conventions and UX around CLI apps, it's not my intent to give heavy weight usability (for now)--especially since this is something nobody else asked for.

Reply:
> I don't want to write code to merge and dedupe the arrays generated by passing --newPkg and --new-pkg


Well, it's there in the suggestion now; I don't think you'll need to write any new code for that? 


> I didn't really want to write any code to handle both kebab-case and camelCase

The suggestion here is indeed to reduce the special-handling - as it doesn't seem necessary.

>  I also didn't want inconsistent behavior around `--dryRun`.

What's the inconsistent behavior?